packages/react-icons/src/utils/fonts/createFluentFontIcon.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { FluentIconsProps } from '../FluentIconsProps.types';
+import { makeStyles, makeStaticStyles, mergeClasses } from "@griffel/react";
+import { useIconState } from '../useIconState';
+
+import fontFilledTtf from './FluentSystemIcons-Filled.ttf';
+import fontFilledWoff from './FluentSystemIcons-Filled.woff';
+import fontFilledWoff2 from './FluentSystemIcons-Filled.woff2';
+
+import fontRegularTtf from './FluentSystemIcons-Regular.ttf';
+import fontRegularWoff from './FluentSystemIcons-Regular.woff';
+import fontRegularWoff2 from './FluentSystemIcons-Regular.woff2';
+
+import fontOneSizeTtf from './FluentSystemIcons-Resizable.ttf';
+import fontOneSizeWoff from './FluentSystemIcons-Resizable.woff';
+import fontOneSizeWoff2 from './FluentSystemIcons-Resizable.woff2';
+
+export const enum FontFile {
+    Filled = 0,
+    Regular = 1,
+    Resizable = 2
+}
+
+const FONT_FAMILY_MAP = {
+    [FontFile.Filled]: 'FluentSystemIconsFilled',
+    [FontFile.Regular]: 'FluentSystemIconsRegular',
+    [FontFile.Resizable]: 'FluentSystemIcons',
+} as const;
+
+const useStaticStyles = makeStaticStyles(`
+@font-face {
+    font-family: ${FONT_FAMILY_MAP[FontFile.Filled]};
+    src: url(${JSON.stringify(fontFilledWoff2)}) format("woff2"),
+    url(${JSON.stringify(fontFilledWoff)}) format("woff"),
+    url(${JSON.stringify(fontFilledTtf)}) format("truetype");
+}
+@font-face {
+    font-family: ${FONT_FAMILY_MAP[FontFile.Regular]};
+    src: url(${JSON.stringify(fontRegularWoff2)}) format("woff2"),
+    url(${JSON.stringify(fontRegularWoff)}) format("woff"),
+    url(${JSON.stringify(fontRegularTtf)}) format("truetype");
+}
+@font-face {
+    font-family: ${FONT_FAMILY_MAP[FontFile.Resizable]};
+    src: url(${JSON.stringify(fontOneSizeWoff2)}) format("woff2"),
+    url(${JSON.stringify(fontOneSizeWoff)}) format("woff"),
+    url(${JSON.stringify(fontOneSizeTtf)}) format("truetype");
+}
+`)
+
+const useRootStyles = makeStyles({
+    root: {
+        display: 'inline-block',
```

Comment:
The SVG icon sets `display: 'inline'`. Should we use the same here?

Reply:
I don't think so. It will still be treated as `inline` externally, but `inline-block` allows it to better control its own dimensions with the `line-height`